src/value.c

```diff
@@ -957,6 +959,110 @@ json_t *json_null(void)
 }
 
 
+/*** raw ***/
+json_t *json_raw_nocheck(const char *value)
+{
+    if(!value)
+        return NULL;
+
+    return string_create(value, strlen(value), 0, JSON_RAW);
+}
+
+json_t *json_rawn_nocheck(const char *value, size_t len)
+{
+    return string_create(value, len, 0, JSON_RAW);
+}
+
+json_t *json_raw(const char *value)
+{
+    if(!value)
+        return NULL;
+
+    return json_rawn(value, strlen(value));
+}
+
+json_t *json_rawn(const char *value, size_t len)
+{
+    if(!value || !utf8_check_string(value, len, 1))
+        return NULL;
+
+    return json_rawn_nocheck(value, len);
+}
```

Comment:
Should we check that the raw value starts with a digit or one of `t`, `f`, `n`, `"`, `[`, `{`? Or should there even be a validation step that involves `json_loads` with `JSON_DECODE_ANY` flag on?

If not, should funtions for creating `JSON_RAW` values be private, at least in the beginning, so that it would be harder for users to create invalid JSON accidentally?

Reply:
This of course assumes that there will be a decoding flag to e.g. decode all numbers as `JSON_RAW`.

Reply:
You've given me the thought that `json_dump_raw` might be useful - something that works the same as json_dumps but returns a json_t containing a raw value.  Then json_raw constructors could be made private but end users could still create arbitrary raw objects and link them to other json_t objects/arrays.

I'm not sure about having these functions perform validation.  I guess for now I'm leaning towards making the create/set functions private?

Reply:
I've dropped the json_raw constructors and setters, replaced with `json_dump_raw` and `json_dump_raw_new`.  The point of `json_dump_raw_new` is to support constructs such as:
```c
json_t *raw = json_dump_raw_new(json_pack("{}"), 0);
```

I think the 'create from string' functions should not be created until needed by `json_loads`, and at that point it will be up to json_loads to validate input even when flags tell it to keep certain input raw.